test/ziggurat/config_test.clj

```diff
@@ -157,3 +159,76 @@
+                                  :key-serializer-class               "key-serializer"
+                                  :retries-config                     5
+                                  :session-timeout-ms-config          4000
+                                  :stream-threads-count               4
+                                  :value-deserializer-class-config    "value-deserializer"
+                                  :value-serializer-class             "value-serializer"}
+          props                   (build-all-config-properties config-map)
+          auto-offset-reset       (.getProperty props "auto.offset.reset")
+          auto-commit-interval-ms (.getProperty props "auto.commit.interval.ms")
+          group-id                (.getProperty props "group.id")
+          replication-factor      (.getProperty props "replication.factor")
+          default-api-timeout-ms  (.getProperty props "default.api.timeout.ms")
+          key-deserializer        (.getProperty props "key.deserializer")
+          key-serializer          (.getProperty props "key.serializer")
+          session-timeout-ms      (.getProperty props "session.timeout.ms")
+          num-stream-threads      (.getProperty props "num.stream.threads")
+          retries                 (.getProperty props "retries")
+          value-deserializer      (.getProperty props "value.deserializer")
+          value-serializer        (.getProperty props "value.serializer")]
+      (is (= auto-offset-reset "latest"))
+      (is (= auto-commit-interval-ms "20000"))
+      (is (= replication-factor "2"))
+      (is (= default-api-timeout-ms "3000"))
+      (is (= key-deserializer "key-deserializer"))
+      (is (= key-serializer "key-serializer"))
+      (is (= session-timeout-ms "4000"))
+      (is (= num-stream-threads "4"))
+      (is (= retries "5"))
+      (is (= value-deserializer "value-deserializer"))
+      (is (= value-serializer "value-serializer"))
+      (is (= group-id "foo"))))
+  (testing "non kafka config keys should not be in Properties"
+    (let [config-map {:consumer-type                 :joins
+                      :producer                      {:foo "bar"}
+                      :channels                      {:bar "foo"}
+                      :oldest-processed-message-in-s 10
+                      :origin-topic                  "origin"
+                      :input-topics                  [:foo :bar]
+                      :join-cfg                      {:foo "foo" :bar "bar"}
+                      :thread-count                  7
+                      :poll-timeout-ms-config        10000}
+          props      (build-all-config-properties config-map)]
+      (map (fn [[k _]]
+             (let [string-key (str/replace (name k) #"-" ".")
+                   not-found  "NOT FOUND!"
+                   v          (.getProperty props string-key not-found)]
+               (is (= v not-found))))
```

Comment:
Is this getting evaluated. Because map creates a lazy sequence. You can use `doall` with map to forcefully yield the lazy sequence.

Reply:
done